File: go-server/models/subject-history.go
package models

// SubjectHistory holds an edit copy of the subject
type SubjectHistory struct {
	Model
	Editor             string `gorm:"not null" json:"editor"`
	PersonName         string `gorm:"not null" json:"personName"`
	CostumeDescription string `gorm:"not null" json:"costumeDescription"`
	SubjectID          uint   `gorm:"not null" json:"subjectId"`
}

File: graphql-server/src/db/models/subject_history.js
'use strict';
module.exports = (sequelize, DataTypes) => {
  var subject_history = sequelize.define(
    'subject_history',
    {
      editor: DataTypes.STRING,
      personName: DataTypes.STRING,
      costumeDescription: DataTypes.STRING,
      subjectId: DataTypes.INTEGER,
    },
    {},
  );
  subject_history.associate = function(models) {
    // associations can be defined here
    subject_history.belongsTo(models.subject, { foreignKey: 'subjectId' });
    subject_history.belongsTo(models.user, { foreignKey: 'editor' });
  };
  return subject_history;
};
